lib/models/MatrixAction.js

```diff
@@ -26,6 +30,47 @@ function MatrixAction(type, text, htmlText, timestamp) {
     this.htmlText = htmlText;
     this.ts = timestamp || 0;
 }
+
+MatrixAction.prototype.formatMentions = function(nickUserIdMap) {
+    const regexString = "(" +
+        Object.keys(nickUserIdMap).map((value) => escapeStringRegexp(value)).join("|")
+        + ")";
+    const userRegex = new RegExp(regexString, "igm");
+    const matched = new Set(); // lowercased nicknames we have matched already.
+    let match;
+    let i = 0;
+    while ((match = userRegex.exec(this.text)) !== null && i < MAX_MATCHES) {
```

Comment:
It'd be nice to have this loop not require so much setup. A `for` loop could work too:

```
for (let i = 0; i < MAX_MATCHES && (match = userRegex.exec(this.text)) !== null) {
```